src/Host/Protocol/Test/UserProfileServicePipe/UserProfileServicePipeTest.cs

```diff
@@ -0,0 +1,140 @@
+                byte[] data = Encoding.Unicode.GetBytes(input);
+
+                await client.WriteAsync(data, 0, data.Length, ct);
+                await client.FlushAsync(ct);
+
+                byte[] responseRaw = new byte[1024];
+                var bytesRead = await client.ReadAsync(responseRaw, 0, responseRaw.Length, ct);
+                jsonResp = Encoding.Unicode.GetString(responseRaw, 0, bytesRead);
+            }
+            return JsonConvert.DeserializeObject<UserProfileResultMock>(jsonResp);
+        }
+
+        private async Task CreateProfileTestRunnerAsync(IUserProfileServices creator, string input, bool isValidParse, bool isValidAccount, bool isExistingAccount, int serverTimeOut, int clientTimeOut, bool isFuzzTest = false) {
+            ManualResetEventSlim testDone = new ManualResetEventSlim(false);
+            Task.Run(async () => {
+                try {
+                    if (isFuzzTest) {
+                        try {
+                            await RUserProfileCreator.CreateProfileAsync(serverTimeOutms: serverTimeOut, clientTimeOutms: clientTimeOut, userProfileService: creator);
+                        } catch (JsonReaderException) {
+                            // expecting JSON parsing to fail
+                            // JSON parsing may fail due to randomly generated strings as input.
+                        }
+                    } else {
+                        if (isValidParse) {
+                            Func<Task> f = async () => await RUserProfileCreator.CreateProfileAsync(serverTimeOutms: serverTimeOut, clientTimeOutms: clientTimeOut, userProfileService: creator);
+                            f.ShouldNotThrow();
+                        } else {
+                            Func<Task> f = () => RUserProfileCreator.CreateProfileAsync(serverTimeOutms: serverTimeOut, clientTimeOutms: clientTimeOut, userProfileService: creator);
+                            await f.ShouldThrowAsync<Exception>();
+                        }
+                    }
+                } finally {
+                    testDone.Set();
+                }
+            }).DoNotWait();
+
+            using (CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(clientTimeOut))) {
+                UserProfileResultMock result = await CreateProfileClientTestWorkerAsync(input, cts.Token);
+                if (isFuzzTest) {
+                    // fuzz test parsing succeeded, the creator always fails for this test.
+                    result?.Error.Should().Be(13);
+                } else {
+                    if (isValidParse) {
+                        result.Error.Should().Be((uint)(isValidAccount ? 0 : 13));
+                        result.ProfileExists.Should().Be(isExistingAccount);
+                    } else {
+                        result.Should().BeNull();
+                    }
+                }
+            }
+
+            testDone.Wait();
```

Comment:
See explanation below for this. we can put a time out here. Note, if this test ends without the server disconnecting, the next test will certainly fail. This wait is needed to ensure all threads are done.